code/modules/mafia/controller.dm

```diff
@@ -0,0 +1,529 @@
+/datum/mafia_controller
+	var/list/all_roles = list()
+	var/list/player_role_lookup = list() //This only exists to speed up role retrieval
+	var/phase = MAFIA_PHASE_SETUP
+	var/turn = 0
+
+	var/player_old_bodies = list() //We send players back to their bodies after
+
+	var/first_day_phase_period = 20 SECONDS //say hi to everyone, no voting so no discussion
+	var/day_phase_period = 1 MINUTES //talk with others about the night
+	var/voting_phase_period = 30 SECONDS //vote someone to get put on trial
+	var/judgement_phase_period = 30 SECONDS //defend yourself! don't get lynched! sometimes skipped
+	var/judgement_lynch_period = 5 SECONDS //just watch a guy die (or live but lol)
+	var/night_phase_period = 45 SECONDS //mafia talk and pick someone to kill
+	var/victory_lap_period = 20 SECONDS //just waiting around sayin' woop woop
+
+	var/list/current_setup_text //Redable list of roles in current game
+
+	var/game_id //Used to sync all parts - signup boards, spawns, night curtains, defaults to "mafia"
+	var/list/signed_up = list()
+
+	var/player_outfit = /datum/outfit/mafia //todo some fluffy outfit
```

Comment:
👀

Reply:
i kinda like the grey generic outfits people get. i might just keep it.